test/hotspot/jtreg/compiler/lib/verify/Verify.java

```diff
@@ -71,107 +109,176 @@ private static void checkEQ(Object a, Object b, String context) {
-            case char[]    x -> checkEQimpl(x, (char[])b,                  context);
-            case short[]   x -> checkEQimpl(x, (short[])b,                 context);
-            case int[]     x -> checkEQimpl(x, (int[])b,                   context);
-            case long[]    x -> checkEQimpl(x, (long[])b,                  context);
-            case float[]   x -> checkEQimpl(x, (float[])b,                 context);
-            case double[]  x -> checkEQimpl(x, (double[])b,                context);
-            case MemorySegment x -> checkEQimpl(x, (MemorySegment) b,      context);
+            case Object[]  x -> checkEQimpl(x, (Object[])b,                 field, aParent, bParent);
+            case Byte      x -> checkEQimpl(x, ((Byte)b).byteValue(),       field, aParent, bParent);
+            case Character x -> checkEQimpl(x, ((Character)b).charValue(),  field, aParent, bParent);
+            case Short     x -> checkEQimpl(x, ((Short)b).shortValue(),     field, aParent, bParent);
+            case Integer   x -> checkEQimpl(x, ((Integer)b).intValue(),     field, aParent, bParent);
+            case Long      x -> checkEQimpl(x, ((Long)b).longValue(),       field, aParent, bParent);
+            case Float     x -> checkEQimpl(x, ((Float)b).floatValue(),     field, aParent, bParent);
+            case Double    x -> checkEQimpl(x, ((Double)b).doubleValue(),   field, aParent, bParent);
+            case Boolean   x -> checkEQimpl(x, ((Boolean)b).booleanValue(), field, aParent, bParent);
+            case byte[]    x -> checkEQimpl(x, (byte[])b,                   field, aParent, bParent);
+            case char[]    x -> checkEQimpl(x, (char[])b,                   field, aParent, bParent);
+            case short[]   x -> checkEQimpl(x, (short[])b,                  field, aParent, bParent);
+            case int[]     x -> checkEQimpl(x, (int[])b,                    field, aParent, bParent);
+            case long[]    x -> checkEQimpl(x, (long[])b,                   field, aParent, bParent);
+            case float[]   x -> checkEQimpl(x, (float[])b,                  field, aParent, bParent);
+            case double[]  x -> checkEQimpl(x, (double[])b,                 field, aParent, bParent);
+            case boolean[] x -> checkEQimpl(x, (boolean[])b,                field, aParent, bParent);
+            case MemorySegment x -> checkEQimpl(x, (MemorySegment) b,       field, aParent, bParent);
+            case Exception x -> checkEQimpl(x, (Exception) b,               field, aParent, bParent);
             default -> {
-                System.err.println("ERROR: Verify.checkEQ failed: type not supported: " + ca.getName());
-                print(a, "a " + context);
-                print(b, "b " + context);
-                throw new VerifyException("Object array type not supported: " + ca.getName());
+                if (ca.getName().startsWith("jdk.incubator.vector") && ca.getName().contains("Vector")) {
```

Comment:
Might be worth to extract this case to own methods and structure it like this to reduce the size of the method:
```
if (vectorClass()) {
    checkEQForVectorAPIClass();
} else {
    checkEQdispatch();
}
```

Reply:
Refactored it :)